analyzers/src/SonarAnalyzer.Common/Rules/Utilities/MetricsAnalyzerBase.cs

```diff
@@ -41,7 +41,7 @@ protected sealed override MetricsInfo CreateMessage(SyntaxTree syntaxTree, Seman
             var complexity = metrics.Complexity;
             var metricsInfo = new MetricsInfo
             {
-                FilePath = syntaxTree.FilePath,
+                FilePath = syntaxTree.GetRoot().GetMappedFilePathFromRoot(),
```

Comment:
We are replacing `syntaxTree.FilePath` with `syntaxTree.GetRoot().GetMappedFilePathFromRoot()` a bit everywhere. 

I wonder if we ever call `GetMappedFilePathFromRoot` on something different from the root of the `SyntaxTree`, and if we don't if it does make sense to define the extension method as an extension of `SyntaxRoot` directly, instead of the `SyntaxNode`.

Moreover, the name of the extension method refers to the root, even though the extension method applies to any `SyntaxNode`, and yet the method falls back to `root.SyntaxTree.FilePath`, which to me is an indication that the extension method is probably better placed at `SyntaxTree` level, as that way you go down in the chain of dependencies, not up.

WDYT?

Reply:
Is there a `SyntaxRoot` class? :hushed:

Reply:
You mean `SyntaxTree` probably. Indeed, we could have something more general or less error-prone. The problem is that we don't have the tree everywhere. We would have to take it from the node (not a big deal). We would avoid some operations by using directly the node.

Let's handle this in another PR.

Reply:
Yes, I meant `SyntaxTree`. I did a quick search and I only found calls of the form `syntaxTree.GetRoot().GetMappedFilePathFromRoot()`, i.e. always starting from the `syntaxTree`.
Maybe I have missed some cases there. It surely can go to another PR.